tools/src/launcher/Main.hs

```diff
@@ -128,13 +126,23 @@ data NodeData = NodeData
     , ndArgs    :: ![Text]
     , ndLogPath :: Maybe FilePath }
 
--- | Updater path, args, windows runner path, archive path
-data UpdaterData = UpdaterData
-    { udPath        :: !FilePath
-    , udArgs        :: ![Text]
-    , udWindowsPath :: Maybe FilePath
-    , udArchivePath :: Maybe FilePath
-    }
+data UpdaterData = WindowsUpdaterData WinUD | OSXUpdaterData OsxUD | UnsupportedOSUpdateData deriving (Generic)
+
+instance FromJSON UpdaterData
+instance FromJSON WinUD
+instance FromJSON OsxUD
```

Comment:
Can you use `genericParseJSON defaultOptions` in these instances please? `Serokell.Util.defaultOptions` are different from default options in `aeson` and they are more suitable for our code.

Reply:
Also previously `updaterArgs` were optional, but now they are not. You can fix it using the trick which was used for `LauncherOptions`.

Reply:
Did you mean Serokell.Aeson.Options (defaultOptions)? Because I can`t see defaultOptions 
 in Serokell.Util. 
>Also previously updaterArgs were optional

What do you mean by optional? It was a [Text]: [] for Windows and ["--FW"] for OS X. I thought it stayed untouched. And I don`t understand, why launcher-config.yaml looks like:
```
# Arguments to be passed to the updater.
updaterArgs:
  - "dir"
  - "binaries"
```

Reply:
I think I already got it. In case you want to answer.